serveradmin/serverdb/query_materializer.py

```diff
@@ -113,24 +139,24 @@ def _add_attributes(self, servers_by_type):
                 for attribute in attributes:
                     self._add_supernet_attribute(attribute, (
                         s for st in self._servertypes_by_attribute[attribute]
-                        for s in servers_by_type[st]
+                        for s in servers_by_type[st.pk]
                     ))
             elif key == 'reverse':
                 reversed_attributes = {
                     a.reversed_attribute: a for a in attributes
                 }
                 for sa in ServerRelationAttribute.objects.filter(
                     value_id__in=self._server_attributes.keys(),
-                    _attribute__in=reversed_attributes.keys(),
-                ).select_related('server'):
+                    attribute__in=reversed_attributes.keys(),
+                ).select_related():
```

Comment:
Why do we not need the select_related('server') here anymore?

Reply:
`select_related()` without any arguments joins all direct relations to the query.  I had abused it to access the attribute objects.  This is wrong though.  Attributes should be passed there as objects already.  I am putting the `'server'` argument back there, and concentrating on reorganizing things more to preserve the Attribute objects.